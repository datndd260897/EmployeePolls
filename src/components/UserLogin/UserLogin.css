/* Updated styles for a modern look */
.login-container {
  max-width: 400px;
  margin: 50px auto;
  padding: 30px;
  border: 1px solid #ccc;
  border-radius: 10px;
  box-shadow: 0 0 15px rgba(0, 0, 0, 0.2);
  background-color: #ffffff;
}

.login-container h2 {
  margin-bottom: 20px;
  color: #2c3e50;
  text-align: center;
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}

.form-group {
  margin-bottom: 15px;
}

.form-group label {
  display: block;
  margin-bottom: 5px;
  color: #34495e;
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}

.form-group input {
  width: 100%;
  padding: 10px;
  box-sizing: border-box;
  border: 1px solid #ddd;
  border-radius: 5px;
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}

.error {
  color: red;
  margin-bottom: 15px;
  text-align: center;
}

button {
  width: 100%;
  padding: 12px;
  background-color: #e74c3c;
  border: none;
  color: white;
  font-size: 16px;
  cursor: pointer;
  border-radius: 5px;
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}

button:hover {
  background-color: #c0392b;
}
