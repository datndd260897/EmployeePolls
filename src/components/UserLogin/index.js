import UserLogin from './UserLogin';
import './UserLogin.css';

export default UserLogin;
