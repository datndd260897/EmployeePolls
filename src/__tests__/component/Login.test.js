import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import UserLogin from '../../components/UserLogin';

const mockStore = configureStore([]);
const initialState = {
  users: {
    foo: {
      id: 'foo',
      password: 'abc123',
    },
  },
};

const store = mockStore(initialState);

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
  useLocation: () => ({
    state: { from: '/' },
  }),
}));

jest.mock('../../actions/authedUser', () => ({
  login: jest.fn().mockReturnValue({ type: 'LOGIN' }),
}));

describe('UserLogin Component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <UserLogin />
        </Router>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('shows an error message with invalid password', () => {
    render(
      <Provider store={store}>
        <Router>
          <UserLogin />
        </Router>
      </Provider>
    );

    fireEvent.change(screen.getByLabelText('Username'), {
      target: { value: 'foo' },
    });
    fireEvent.change(screen.getByLabelText('Password'), {
      target: { value: 'abc321' },
    });

    fireEvent.click(screen.getByRole('button', { name: 'UserLogin' }));

    expect(
      screen.getByText(
        'The username or password is incorrect. Please try again.'
      )
    ).toBeInTheDocument();
  });
});
