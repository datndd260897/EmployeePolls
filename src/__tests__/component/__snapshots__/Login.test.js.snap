// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="login-container"
  >
    <h2>
      Login
    </h2>
    <div
      class="error"
    />
    <form>
      <div
        class="form-group"
      >
        <label
          for="username"
        >
          Username
        </label>
        <input
          id="username"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          for="password"
        >
          Password
        </label>
        <input
          id="password"
          required=""
          type="password"
          value=""
        />
      </div>
      <button
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</DocumentFragment>
`;
